manager/volume.go

```diff
@@ -409,14 +408,28 @@ func (m *VolumeManager) checkMigratingEngineSyncSnapshots(vol *longhorn.Volume)
 		return false, fmt.Errorf("failed to find the migrating engine for volume %v", vol.Name)
 	}
 
-	if !reflect.DeepEqual(oldEngine.Status.Snapshots, migratingEngine.Status.Snapshots) {
+	if !hasSameKeys(oldEngine.Status.Snapshots, migratingEngine.Status.Snapshots) {
 		logrus.Debugf("Volume migration (%v) is in progress for synchronizing snapshots", vol.Name)
 		return false, nil
 	}
 
 	return true, nil
 }
 
+func hasSameKeys(map1, map2 map[string]*longhorn.SnapshotInfo) bool {
```

Comment:
NIT: I would prefer to see a comment here explaining why this function is needed (e.g. map[string]*longhorn.SnapshotInfo can't be compared directly because created field might be different).

Reply:
Added the comment in the above function. Thank you!